sql/src/test/java/org/apache/druid/sql/calcite/CalciteWindowQueryTest.java

```diff
@@ -91,119 +88,117 @@ public static Object parametersForWindowQueryTest() throws Exception
+      for (int i = 0; i < outputSignature.size(); ++i) {
+        types[i] = outputSignature.getColumnType(i).get();
+        Assert.assertEquals(types[i], results.signature.getColumnType(i).get());
+      }
 
-    Function<Object, String> jacksonToString = value -> {
-      try {
-        return queryJackson.writeValueAsString(value);
+      maybeDumpActualResults(results.results);
+      for (Object[] result : input.expectedResults) {
+        for (int i = 0; i < result.length; i++) {
+          // Jackson deserializes numbers as the minimum size required to
+          // store the value. This means that
+          // Longs can become Integer objects and then they fail equality
+          // checks. We read the expected
+          // results using Jackson, so, we coerce the expected results to the
+          // type expected.
+          if (result[i] != null) {
+            if (result[i] instanceof Number) {
+              switch (types[i].getType()) {
+                case LONG:
+                  result[i] = ((Number) result[i]).longValue();
+                  break;
+                case DOUBLE:
+                  result[i] = ((Number) result[i]).doubleValue();
+                  break;
+                case FLOAT:
+                  result[i] = ((Number) result[i]).floatValue();
+                  break;
+                default:
+                  throw new ISE("result[%s] was type[%s]!?  Expected it to be numerical", i, types[i].getType());
+              }
+            }
+          }
+        }
       }
-      catch (JsonProcessingException e) {
-        throw new RE(e);
+      assertResultsValid(filename, input.expectedResults, results);
+    }
+
+    private void maybeDumpActualResults(List<Object[]> results) throws Exception
+    {
+      for (Object[] row : results) {
+        System.out.println("  - " + queryJackson.writeValueAsString(row));
       }
-    };
+    }
+  }
+
+  @Test
+  @SuppressWarnings("unchecked")
+  public void windowQueryTest() throws Exception
+  {
+    TestCase testCase = new TestCase(filename);
 
-    assumeThat(input.type, Matchers.not(TestType.failingTest));
+    assumeThat(testCase.getType(), Matchers.not(TestType.failingTest));
 
-    if (input.type == TestType.operatorValidation) {
+    if (testCase.getType() == TestType.operatorValidation) {
       testBuilder()
           .skipVectorize(true)
-          .sql(input.sql)
+          .sql(testCase.getSql())
           .queryContext(ImmutableMap.of(PlannerContext.CTX_ENABLE_WINDOW_FNS, true,
               QueryContexts.ENABLE_DEBUG, true))
-          .addCustomVerification(QueryVerification.ofResults(results -> {
-            if (results.exception != null) {
-              throw new RE(results.exception, "Failed to execute because of exception.");
-            }
-
-            Assert.assertEquals(1, results.recordedQueries.size());
-            // 2 tests are failing at this moment on this check
```

Comment:
this comment seems to have been dropped in the refactor - is that intended? I still see that those tests are in `operatorValidation` type.

Reply:
I think this comment was originally misplaced when it was added - I believe notices like this should be next to the testcase; or somewhere around the `ignore` ; so that it gets removed when it gets enabled back.

`operatorValidation` is kinda like the "enable" mode for these tests  - I see only one of the old tests (wikipediaAggregationsMultipleOrderingDesc.sqlTest) to be marked as `failingTest` ; and that one is reporting an issue because there is a `DESC` in the query which is not supported..actually that test was never enabled; ever since it was added

```
dev@windowing-fixes:~/druid$ grep -r failing sql/src/test/resources/calcite/tests/window/*
sql/src/test/resources/calcite/tests/window/no_grouping.sqlTest:type: "failingTest"
sql/src/test/resources/calcite/tests/window/wikipediaAggregationsMultipleOrderingDesc.sqlTest:type: "failingTest"
```

Reply:
Thanks for the clarification @kgyrtkirk